packages/flutter_tools/lib/src/project.dart

```diff
@@ -692,6 +692,39 @@ class AndroidProject extends FlutterProjectPlatform {
     gradle.updateLocalProperties(project: parent, requireAndroidSdk: false);
   }
 
+  /// Returns true if the current version of the Gradle plugin is supported.
+  bool isSupportedVersion() {
```

Comment:
I placed these methods on the AndroidProject since they mostly behave like they are already members of it. Also made the testing setup easier